### PR TITLE
RavenDB-14414: add failover when copying pal file

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Sparrow.Server/Platform/Pal.cs [hunk 1]
```diff
@@ -99,6 +98,33 @@ static Pal()
                 PalHelper.ThrowLastError(rc, errorCode, "Cannot get system information");
         }
 
+        private static bool CopyPalFile(string toFilename, string fromFilename, out Exception exception)
```
Comment (ayende) [inline src/Sparrow.Server/Platform/Pal.cs]:
```suggestion
        private static Exception CopyPalFile(string toFilename, string fromFilename, out Exception exception)
```

File: src/Sparrow.Server/Platform/Pal.cs [hunk 2]
```diff
@@ -45,28 +46,26 @@ static Pal()
...
 
-                if (copy)
-                    File.Copy(fromFilename, toFilename, overwrite: true);
-            }
-            catch (IOException e)
-            {
-                throw new IOException(
-                    $"Cannot copy {fromFilename} to {toFilename}, make sure appropriate {toFilename} to your platform architecture exists in Raven.Server executable folder",
-                    e);
+                throw e;
```
Comment (ayende) [inline src/Sparrow.Server/Platform/Pal.cs]: Here you lost the exception trace, `throw new InvalidOperationException('msg', e);`

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Sparrow.Server/Platform/Pal.cs (hunk 1 above)
Comment (garayx) [reply src/Sparrow.Server/Platform/Pal.cs]: done

File: src/Sparrow.Server/Platform/Pal.cs (hunk 2 above)
Comment (garayx) [reply src/Sparrow.Server/Platform/Pal.cs]: done

**Review by ppekrol — APPROVED** [review]

**Pull request merged**